src/agent/pythd/adapter/local.rs

```diff
@@ -0,0 +1,110 @@
+// The Local Store stores a copy of all the price information this local publisher
+// is contributing to the network. The Exporters will then take this data and publish
+// it to the networks.
+use {
+    super::{
+        Adapter,
+        AdapterApi,
+        PriceIdentifier,
+    },
+    crate::agent::metrics::PriceLocalMetrics,
+    anyhow::{
+        anyhow,
+        Result,
+    },
+    chrono::NaiveDateTime,
+    prometheus_client::registry::Registry,
+    pyth_sdk_solana::state::PriceStatus,
+    slog::Logger,
+    solana_sdk::bs58,
+    std::collections::HashMap,
+    tokio::sync::RwLock,
+};
+
+#[derive(Clone, Debug)]
+pub struct PriceInfo {
+    pub status:    PriceStatus,
+    pub price:     i64,
+    pub conf:      u64,
+    pub timestamp: NaiveDateTime,
+}
+
+impl PriceInfo {
+    /// Returns false if any non-timestamp fields differ with `other`. Used for last published state comparison in exporter.
+    pub fn cmp_no_timestamp(&self, other: &Self) -> bool {
+        // Prevent forgetting to use a new field if we expand the type.
+        #[deny(unused_variables)]
+        let Self {
+            status,
+            price,
+            conf,
+            timestamp: _,
+        } = self;
+
+        status == &other.status && price == &other.price && conf == &other.conf
+    }
+}
+
+pub struct Store {
+    prices:  RwLock<HashMap<PriceIdentifier, PriceInfo>>,
```

Comment:
not for this one as it's restructing, but in the future DashMap would make it better because update seems to be for each price feed.